### ISSUE TITLE
svg.c: missing EOF checking

### ISSUE THREAD

**Issue by pgajdos** [ImageMagick/ImageMagick#1089]
Testcase from issue #539 does hang for me with 7.0.7-29 and 6.9.9-40. If I debugged correctly, the key is here:
https://github.com/ImageMagick/ImageMagick/blob/master/coders/svg.c#L3043
(and perhaps also here:
https://github.com/ImageMagick/ImageMagick/blob/master/coders/svg.c#L3299)

where `ReadBlob()` returns `EOF` for this testcase.
Thanks!

**Comment by urban-warrior** [comment]
Unfortunately we cannot reproduce the problem.  Post the command you are using that produces the hang for you.  Add -verbose to the command line to help determine which SVG renderer you are using.  ImageMagick supports inkscape, rsvg, and msvg.

**Comment by pgajdos** [comment]
Thanks for the reply.

Command:
```
$ identify imagemagick-heap-buffer-overflow-4
[run long/hang]
```
We are using rsvg:
```
$ ldd /usr/lib64/ImageMagick-7.0.7/modules-7_Q16HDRI5/coders/svg.so | grep svg
	librsvg-2.so.2 => /usr/lib64/librsvg-2.so.2 (0x00007f07ee039000)
$
```
Sorry for the typo in initial comment. I am testing with 7.0.7-28.

**Comment by pgajdos** [comment]
Yes, the as far as I tested with 6.9.10-6 and 7.0.8-6.

**Comment by pgajdos** [comment]
Yes, the problem is solved as far as I tested with 6.9.10-6 and 7.0.8-6.
